greenplumpython/dataframe.py

```diff
@@ -170,66 +170,83 @@ def __getitem__(self, _):
+            "|".join(
+                [
+                    " {:{}} ".format(col if col is not None else "", width)
+                    for col, width in widths.items()
+                ]
+            )
+            + "\n"
+        )
+        df_string += line("+")
+        for row in contents:
+            df_string += (
+                "|".join(
                     [
-                        " {:{}} ".format(col, width[idx]) + ("|" if idx < col_number - 1 else "")  # type: ignore
-                        for idx, col in enumerate(ret[0])  # type: ignore
+                        (" {:{}} ").format(
+                            "{}".format(
+                                ""
+                                if col_name is None
+                                else row[col_name]
+                                if isinstance(row[col_name], list)
+                                else ("{:{}}").format(
+                                    row[col_name] if row[col_name] is not None else "",
+                                    widths[col_name],
+                                )
+                            ),
+                            widths[col_name],
+                        )
+                        for col_name in widths
                     ]
                 )
-                + "\n"
-            )
-            # Dividing line below DataFrame header.
-            repr_string += (
-                "+".join(
-                    [" {:{}} ".format("-" * width[idx], width[idx]) for idx in range(col_number)]
-                )
-                + "\n"
-            )
-            # DataFrame contents.
-            for row in ret:
-                content = [row[c] for c in row]  # type: ignore
-                for idx, c in enumerate(content):  # type: ignore
-                    if isinstance(c, list):
-                        repr_string += (" {:{}} ").format("{}".format(c), width[idx])  # type: ignore
-                        repr_string += "|" if idx < col_number - 1 else ""
-                    else:
-                        repr_string += (" {:{}} ").format(c if c is not None else "", width[idx])  # type: ignore
-                        repr_string += "|" if idx < col_number - 1 else ""
-                repr_string += "\n"
-        return repr_string
+            ) + "\n"
+        df_string += line("-")
+        df_string += row_num_string
+        return df_string
 
-    def _repr_html_(self):
+    def _repr_html_(self) -> str:
         """:meta private:"""
         repr_html_str = ""
         ret = list(self)
+        if len(ret) == 0:
+            return "<table><tr><th></th></tr></table>"
         if len(ret) != 0:
             repr_html_str = "<table>\n"
             repr_html_str += "\t<tr>\n"
-            repr_html_str += ("\t\t<th>{:}</th>\n" * len(ret[0])).format(
-                *((ret[0]))  # type: ignore
-            )
+            repr_html_str += ("\t\t<th>{:}</th>\n" * len(ret[0])).format(*((ret[0])))
```

Comment:
else here maybe better?
if not ret:
    return ///
else:
   do/ // 
   return